lib/spack/spack/environment.py

```diff
@@ -1166,7 +1211,18 @@ def install(self, user_spec, concrete_spec=None, **install_args):
 
     def _install(self, spec, **install_args):
         # "spec" must be concrete
-        spec.package.do_install(**install_args)
+        package = spec.package
+
+        for dev_spec, path in self.dev_specs.items():
+            if spec.satisfies(dev_spec, strict=True):
```

Comment:
You'll probably also need to explicitly disable the cache when hitting this condition as well.

Reply:
I don't need it in the other place, since it calls through to here. I made the change here.